test/framework/e2e_framework.go

```diff
@@ -35,98 +37,64 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-const (
-	defaultControlPlaneContext = "kind-k8ssandra-0"
-)
-
 type E2eFramework struct {
 	*Framework
-
-	nodeToolStatusUN *regexp.Regexp
-	nodeToolStatusDN *regexp.Regexp
 }
 
-func NewE2eFramework(t *testing.T) (*E2eFramework, error) {
-	configFile, err := filepath.Abs("../../build/kubeconfig")
-	if err != nil {
-		return nil, err
-	}
-
-	if _, err = os.Stat(configFile); err != nil {
-		return nil, err
-	}
+var (
+	nodeToolStatusUN = regexp.MustCompile("UN\\s\\s")
+	nodeToolStatusDN = regexp.MustCompile("DN\\s\\s")
+)
 
+func NewE2eFramework(t *testing.T, configFile, controlPlane string, dataPlanes ...string) (*E2eFramework, error) {
 	config, err := clientcmd.LoadFromFile(configFile)
 	if err != nil {
 		return nil, err
 	}
 
-	controlPlaneContext := ""
-	var controlPlaneClient client.Client
 	remoteClients := make(map[string]client.Client, 0)
 	t.Logf("Using config file: %s", configFile)
 
-	for name, _ := range config.Contexts {
-		clientCfg := clientcmd.NewNonInteractiveClientConfig(*config, name, &clientcmd.ConfigOverrides{}, nil)
-		restCfg, err := clientCfg.ClientConfig()
+	if remoteClient, err := newRemoteClient(config, controlPlane); err != nil {
+		return nil, err
+	} else if remoteClient == nil {
+		return nil, fmt.Errorf("control plane context %s does not exist", controlPlane)
+	} else {
+		remoteClients[controlPlane] = remoteClient
+	}
 
-		if err != nil {
+	var validDataPlanes []string
+	for _, name := range dataPlanes {
+		if remoteClient, err := newRemoteClient(config, name); err != nil {
 			return nil, err
-		}
-
-		remoteClient, err := client.New(restCfg, client.Options{Scheme: scheme.Scheme})
-		if err == nil {
+		} else if remoteClient == nil {
+			t.Logf("ignoring invalid data plane context %v: %v", name, err)
```

Comment:
If the data plane context is invalid, do you think we should return an error here?

Reply:
I was returning an error previously, then I bumped into this situation: I created a single cluster with `make single-up`, then I ran the `single-dc` test. It failed, because by default `dataPlanes` is `k8ssandra-1`, and so the framework attempts to create a client for `k8ssandra-1` (even if it is not required for the test being executed).

So I changed this as a convenience for developers. Otherwise you would have to run single dc tests with `-dataPlanes ''` each time.